app/src/components/features/rules/RulePairs/ResponseRuleResourceTypes/index.tsx

```diff
@@ -0,0 +1,165 @@
+//@ts-ignore
+import { ReactComponent as DesktopIcon } from "assets/icons/desktop.svg";
+import { omit, set } from "lodash";
+import { ResponseRule, ResponseRuleResourceType } from "types/rules";
+import "./ResponseRuleResourceTypes.css";
+
+const DownloadDesktopAppLink: React.FC = () => (
+  <a
+    target="_blank"
+    rel="noreferrer"
+    className="download-desktop-app-link"
+    href={APP_CONSTANTS.LINKS.REQUESTLY_DESKTOP_APP}
+  >
+    <DesktopIcon /> <span>Download desktop app</span>
+  </a>
+);
+
+interface ResponseRuleResourceTypesProps {
+  currentlySelectedRuleData: ResponseRule;
+  responseRuleResourceType: ResponseRuleResourceType;
+  setResponseRuleResourceType: (type: ResponseRuleResourceType) => void;
+}
+
+const ResponseRuleResourceTypes: React.FC<ResponseRuleResourceTypesProps> = ({
+  currentlySelectedRuleData,
+  responseRuleResourceType,
+  setResponseRuleResourceType,
+}) => {
+  const dispatch = useDispatch();
+  const isDesktop = useMemo(isDesktopMode, []);
+  const currentResourceType =
+    currentlySelectedRuleData?.pairs?.[0]?.response?.resourceType;
+
+  const [resourceType, setResourceType] = useState<ResponseRuleResourceType>(
+    currentResourceType ?? ResponseRuleResourceType.UNKNOWN
+  );
+  const [
+    responseTypePopupVisible,
+    setResponseTypePopupVisible,
+  ] = useState<boolean>(false);
+
+  const filters =
+    currentlySelectedRuleData.pairs?.[0].source?.filters?.[0]?.requestPayload;
+
+  const isPopConfirmDisabled =
+    currentResourceType !== ResponseRuleResourceType.GRAPHQL_API ||
+    Object.keys(filters ?? {}).length === 0;
+
+  useEffect(() => {
+    if (currentResourceType) {
+      setResponseRuleResourceType(currentResourceType);
+    }
+  }, [currentResourceType, setResponseRuleResourceType]);
+
+  const updateResourceType = (
+    resourceType: ResponseRuleResourceType,
+    clearGraphqlRequestPayload = false
+  ) => {
+    const pairIndex = 0; // response rule will only have one pair
+    const copyOfCurrentlySelectedRule = JSON.parse(
+      JSON.stringify(currentlySelectedRuleData)
+    );
+
+    let updatedPair = set(
+      copyOfCurrentlySelectedRule.pairs[pairIndex],
+      "response.resourceType",
+      resourceType
+    );
+
+    if (clearGraphqlRequestPayload) {
+      // clear graphql request payload on resource type change
+      updatedPair = omit(updatedPair, ["source.filters[0].requestPayload"]);
+    }
+
+    const updatedRule = {
+      ...currentlySelectedRuleData,
+      pairs: [{ ...updatedPair }],
+    };
+
+    setCurrentlySelectedRule(dispatch, updatedRule, true);
+  };
+
+  const handleOnConfirm = () => {
+    const clearGraphqlRequestPayload =
+      resourceType !== ResponseRuleResourceType.GRAPHQL_API;
+
+    setResponseTypePopupVisible(false);
+    setResponseRuleResourceType(resourceType);
+    updateResourceType(resourceType, clearGraphqlRequestPayload);
+  };
+
+  const handleResourceTypeChange = (type: ResponseRuleResourceType) => {
```

Comment:
If not done, please add analytics events on updating resource type and entering graphql operation name.